### ISSUE TITLE
"upload output" events missing from JSON profile with --experimental_remote_cache_async

### ISSUE THREAD

**Issue by jin** [bazelbuild/bazel#20404]
### Description of the bug:

When enabling `--experimental_remote_cache_async`, the "upload outputs" events are missing from the JSON profiler. This makes the profile somewhat inaccurate, because those work is still being done, but just not as a blocking task post-action.

### Which category does this issue belong to?

Remote Execution

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Run a remote-cache enabled build that triggers CAS uploads with and without `--experimental_remote_cache_async`, and compare the JSON profiles.

```
$ cat without_flag.command.profile | jq -j '.traceEvents | .[] | select(.name == "upload outputs") | .dur/1000000,    " ", .name, "\n"' | sort -nr
48.332118 upload outputs
48.3279 upload outputs
21.403152 upload outputs
4.809376 upload outputs
4.638287 upload outputs
3.023634 upload outputs
2.981836 upload outputs
2.16031 upload outputs
1.602159 upload outputs
1.368337 upload outputs
1.211588 upload outputs
1.210596 upload outputs
1.184084 upload outputs
0.998725 upload outputs
0.865894 upload outputs
0.801442 upload outputs
0.761732 upload outputs
0.730206 upload outputs
0.588362 upload outputs
0.569864 upload outputs
0.478627 upload outputs
0.477778 upload outputs
0.438619 upload outputs
0.330436 upload outputs
0.321471 upload outputs
0.315897 upload outputs
0.314685 upload outputs
0.281717 upload outputs
0.253382 upload outputs
0.251439 upload outputs
jingwen@jingwen-macbookpro Downloads % cat with_flag.command.profile | jq -j '.traceEvents | .[] | select(.name == "upload outputs") | .dur/1000000,    " ", .name, "\n"' | sort -nr
```

### Which operating system are you running Bazel on?

linux

### What is the output of `bazel info release`?

6.2

### Is this a regression? If yes, please try to identify the Bazel commit where the bug was introduced.

I think it's just wasn't implemented - http://google3/third_party/bazel/src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java;l=1382-1419;rcl=586625265 looks like the async steps aren't wrapped in a profiler call, but the synchronous ones are.

**Comment by brentleyjones** [comment]
cc @coeuvre @tjgq

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.3.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.3.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.3.0rc1`. Thanks!
